lib/dry/schema/macros/filled.rb

```diff
@@ -20,11 +29,36 @@ def call(*predicates, **opts, &block)
           end
 
           if opts[:type_spec]
-            value(predicates[0], :filled?, *predicates[1..predicates.size - 1], **opts, &block)
+            if filter?
+              parent.filter(:filled?)
```

Comment:
`parent` could query the child and call `filter` on itself before combining traces

Reply:
Something like this (specs passed):
```
diff --git a/lib/dry/schema/macros/dsl.rb b/lib/dry/schema/macros/dsl.rb
index eeed4d0..e3b6da6 100644
--- a/lib/dry/schema/macros/dsl.rb
+++ b/lib/dry/schema/macros/dsl.rb
@@ -66,9 +66,13 @@ module Dry
         # @return [Macros::Core]
         #
         # @api public
-        def filled(*args, &block)
+        def filled(*args, **opts, &block)
           append_macro(Macros::Filled) do |macro|
-            macro.call(*args, &block)
+            defined = macro.call(*args, **opts, &block)
+
+            filter(:filled?) if opts[:type_spec] && defined.filter?
+
+            defined
           end
         end
 
diff --git a/lib/dry/schema/macros/filled.rb b/lib/dry/schema/macros/filled.rb
index 27ecfb0..fdf85c2 100644
--- a/lib/dry/schema/macros/filled.rb
+++ b/lib/dry/schema/macros/filled.rb
@@ -30,7 +30,6 @@ module Dry
 
           if opts[:type_spec]
             if filter?
-              parent.filter(:filled?)
               value(*predicates, **opts, &block)
             else
               value(predicates[0], :filled?, *predicates[1..predicates.size - 1], **opts, &block)
```

Reply:
that's a nice idea, thanks!